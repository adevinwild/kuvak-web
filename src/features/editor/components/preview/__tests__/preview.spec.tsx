import { act, render, screen } from "@testing-library/react";
import ScreenshotPreview from "../index";
import { readFileSync } from "fs";

function localFileToBlob() {
  const placeholder = readFileSync(__dirname + "/placeholder.png");
  return new Blob([placeholder], { type: "image/png" });
}

const imageUploaded = localFileToBlob();

jest.mock("../../../stores/upload-store", () => {
  return {
    useUploadStore: () => ({
      imageUploaded,
      setImage: jest.fn(),
      reset: jest.fn(),
    }),
  };
});

function setupTests() {
  // Mocking the useUploadStore

  // Mock the `scrollIntoView` method
  const scrollIntoViewMock = jest.fn();
  window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;
}

describe("ScreenshotPreview", () => {
  beforeAll(() => {
    setupTests();
  });

  it("ScreenshotPreview - should render without issues", () => {
    act(() => {
      const { asFragment } = render(<ScreenshotPreview />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it("ScreenshotPreview - should render the container", () => {
    act(() => {
      render(<ScreenshotPreview />);
    });

    const container = screen.getByTestId("screenshot-preview");
    expect(container).toBeInTheDocument();
  });
});
