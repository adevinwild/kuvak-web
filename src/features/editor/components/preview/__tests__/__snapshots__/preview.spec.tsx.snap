// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenshotPreview ScreenshotPreview - should render without issues 1`] = `<DocumentFragment />`;
